### PR TITLE
[exporter/loadbalancing] Change config fields to use snake_case

### PR DESCRIPTION
**Pull request by dmitryax**
Change AWS Cloud map resolver config fields from camelCase to snake_case.

The snake_case is required in OTel Collector config fields. It used to be enforced by tests in cmd/oteltestbedcol, but we had to disable them some time ago. Now, the tests are going to be enforced on every component independently. Hence, the camelCase config fields recently added with the new AWS Cloud Map resolver has to be fixed.

### CODE REVIEW

**Review by songy23 — APPROVED** [review]
cc @jpkrohling

**Comment by astencel-sumo** [comment]
I know it's a pain, but given that this component is in "Beta" stability, shouldn't we do the rename via a feature gate?

**Comment by dmitryax** [comment]
The feature was introduced only in 0.97.0 https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/27588 so I expect it's not widely used yet. 

And I want to unblock https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32328 sooner to avoid any other issues like this merged in.

So I'd rather make it as a breaking change but happy to hear what others think cc @jpkrohling @open-telemetry/collector-contrib-approvers

**Comment by astencel-sumo** [comment]
> The feature was introduced only in 0.97.0 https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/27588 so I expect it's not widely used yet. 

Makes sense to me 🚀

**Review by andrzej-stencel — APPROVED** [review]
The README needs an update as well

**Comment by dmitryax** [comment]
Updated README

File: exporter/loadbalancingexporter/README.md
Comment (crobert-1) [inline exporter/loadbalancingexporter/README.md:75]:
```suggestion
  * `health_status` filter in AWS Cloud Map, you can specify the health status of the instances that you want to discover. The health_status filter is optional and allows you to query based on the health status of the instances. 
```

File: exporter/loadbalancingexporter/README.md
Comment (crobert-1) [inline exporter/loadbalancingexporter/README.md:1]: There's an AWS Cloud Map resolver example below in the README:
```
      awsCloudMap:
        namespace: aws-namespace
        serviceName: aws-otel-col-service-name
```
Should be:
```
      aws_cloud_map:
        namespace: aws-namespace
        service_name: aws-otel-col-service-name

File: exporter/loadbalancingexporter/README.md
Comment (crobert-1) [inline exporter/loadbalancingexporter/README.md:71]:
```suggestion
  * `service_name` The name of the service that you specified when you registered the instance, e.g. `otelcollectors`.  If no `service_name` is specified, this will fail to start the Load Balancer exporter.
```

**Review by crobert-1 — COMMENTED** [review]
Another reference that needs fixed (output of an error message):

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/adae3366228735fd906832e875bef4fdfc9b78a5/exporter/loadbalancingexporter/resolver_aws_cloudmap.go#L30

Some other references that should possibly be updated (log messages for config names):

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/adae3366228735fd906832e875bef4fdfc9b78a5/exporter/loadbalancingexporter/resolver_aws_cloudmap.go#L117-L122

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/adae3366228735fd906832e875bef4fdfc9b78a5/exporter/loadbalancingexporter/loadbalancer.go#L95

**Comment by dmitryax** [comment]
Thanks @crobert-1. Updated

File: exporter/loadbalancingexporter/README.md
Comment (bryan-aguilar) [inline exporter/loadbalancingexporter/README.md:197]:
```suggestion
        service_name: otel-col-service-name
```
Teeny tiny nit. Vendor agnostic right :)

**Review by bryan-aguilar — APPROVED** [review]

**Review by crobert-1 — APPROVED** [review]
I believe this is the last reference, but not necessarily essential:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/adae3366228735fd906832e875bef4fdfc9b78a5/exporter/loadbalancingexporter/loadbalancer.go#L95

File: exporter/loadbalancingexporter/README.md
Comment (dmitryax) [reply exporter/loadbalancingexporter/README.md:197]: Should I change `aws-namespace` as well? It's pretty hard to make AWS Cloud Map resolver cloud agnostic :)

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**
